bgpd/bgp_vty.c

```diff
@@ -6477,10 +6418,18 @@ DEFUN (vpn_policy_rt,
 
 			vpn_leak_prechange(dir, afi, bgp_get_default(), bgp);
 
-			if (bgp->vpn_policy[afi].rtlist[dir])
-				ecommunity_free(
-					&bgp->vpn_policy[afi].rtlist[dir]);
-			bgp->vpn_policy[afi].rtlist[dir] = ecommunity_dup(ecom);
+			if (yes) {
+				if (bgp->vpn_policy[afi].rtlist[dir])
+					ecommunity_free(
+					    &bgp->vpn_policy[afi].rtlist[dir]);
+				bgp->vpn_policy[afi].rtlist[dir] =
+					ecommunity_dup(ecom);
+			} else {
+				if (bgp->vpn_policy[afi].rtlist[dir])
+					ecommunity_free(
+					    &bgp->vpn_policy[afi].rtlist[dir]);
+				bgp->vpn_policy[afi].rtlist[dir] = NULL;
+			}
 
 			vpn_leak_postchange(dir, afi, bgp_get_default(), bgp);
```

Comment:
If the no/delete removes all RTs, isn't postchange operation unnecessary? It bails out if RT is not there way into the code.

Reply:
I'm wary of removing this call here because it would not be obvious (to someone working on the code later) that this optimization can be or has been done. In the interest of avoiding future confusion by preserving a consistent design pattern of "prechange; change; postchange" I would prefer to leave this call in.